### PR TITLE
First network element impact implementation

### PR DESCRIPTION
**Pull request by SlimaneAmar**

### CODE REVIEW

**Comment by flomillot** [comment]
Why "Impact", what does that mean exactly ? Is this a business word ?
"ElementImpact" is not very clear to me and there is no Javadoc to explain it.

File: src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+public class SimpleElementImpact extends AbstractElementImpact {
+    public enum SimpleImpactType {
+        CREATION,
+        MODIFICATION,
+        DELETION
+    }
+
+    private SimpleImpactType impactType;
+
+    private String elementId;
```
Comment (flomillot) [inline src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java:33]: Why elementId and not equipmentId, what's the difference ? Same for elementType

File: src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    private IdentifiableType elementType;
+
+    private Set<String> substationIds;
```
Comment (flomillot) [inline src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java:37]: impactedSubstationIds ?

**Review by geofjamg — APPROVED** [review]

**Comment by flomillot** [comment]
Why not using NetworkModificationResult as return value too on updateModificationGroup endpoint ?

File: src/main/java/org/gridsuite/modification/server/impacts/AbstractElementImpact.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.modification.server.impacts;
+
+/**
+ * @author Slimane Amar <slimane.amar at rte-france.com>
+ */
+public abstract class AbstractElementImpact {
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/impacts/AbstractElementImpact.java]: Removed

File: src/main/java/org/gridsuite/modification/server/impacts/CollectionElementImpact.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import com.powsybl.iidm.network.IdentifiableType;
+import lombok.Builder;
+
+import java.util.Set;
+
+/**
+ * @author Slimane Amar <slimane.amar at rte-france.com>
+ */
+@Builder
+public class CollectionElementImpact extends AbstractElementImpact {
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/impacts/CollectionElementImpact.java]: Removed

File: src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * @author Slimane Amar <slimane.amar at rte-france.com>
+ */
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@Setter
+@Getter
+@Builder
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+@ToString
+public class SimpleElementImpact {
```
Comment (flomillot) [inline src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java:24]: Why "Simple", is there more complicated impacts coming after ?

File: src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java:33]: Currently in discussion with @geofjamg

File: src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java (hunk 5 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java:24]: Yes 
See the description in the US

**Review by flomillot — CHANGES_REQUESTED** [review]
I insist on the fact that the name of the DTO is confusing for me...

"ElementImpact" means in french "impact de l'élément", or here it's the impacted element that we want (l'élement impacté), so it should be "ImpactedElement".

If you really don't want to change it, please add Javadoc at least, because it's really confusing.

File: src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java (hunk 2 above)
Comment (flomillot) [reply src/main/java/org/gridsuite/modification/server/impacts/SimpleElementImpact.java:37]: Cf my comment below

File: src/test/java/org/gridsuite/modification/server/Impacts/ElementImpactTest.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        assertEquals("{\"impactType\":\"MODIFICATION\",\"elementId\":\"loadId\",\"elementType\":\"LOAD\",\"substationIds\":[\"s3\"]}", mapper.writeValueAsString(modificationImpact));
+        assertEquals("{\"impactType\":\"DELETION\",\"elementId\":\"generatorId\",\"elementType\":\"GENERATOR\",\"substationIds\":[\"s4\"]}", mapper.writeValueAsString(deletionImpact));
+
+        NetworkModificationResult result = NetworkModificationResult.builder()
+            .applicationStatus(ApplicationStatus.ALL_OK)
+            .networkImpacts((List<SimpleElementImpact>) impacts)
+            .build();
+        assertEquals(
+            "{\"applicationStatus\":\"ALL_OK\",\"networkImpacts\":[{\"impactType\":\"CREATION\",\"elementId\":\"lineId\",\"elementType\":\"LINE\",\"substationIds\":[\"s1\",\"s2\"]},{\"impactType\":\"MODIFICATION\",\"elementId\":\"loadId\",\"elementType\":\"LOAD\",\"substationIds\":[\"s3\"]},{\"impactType\":\"DELETION\",\"elementId\":\"generatorId\",\"elementType\":\"GENERATOR\",\"substationIds\":[\"s4\"]}],\"impactedSubstationsIds\":[\"s1\",\"s2\",\"s3\",\"s4\"]}",
```
Comment (flomillot) [inline src/test/java/org/gridsuite/modification/server/Impacts/ElementImpactTest.java:54]: Do it a JSON file ?

File: src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java [hunk 7]
```diff
@@ -755,355 +740,165 @@ public void testTombstonedEquipmentInfos() throws Exception {
...
              .andExpect(status().isOk()).andReturn();
-        resultAsString = mvcResult.getResponse().getContentAsString();
-        List<EquipmentDeletionInfos> deletionsDeletionEquipement = mapper.readValue(resultAsString, new TypeReference<>() { });
-        assertThat(deletionsDeletionEquipement.get(0), createMatcherEquipmentDeletionInfos(equipmentDeletionInfos));
-
+        testBranchDeletionImpacts(mvcResult.getResponse().getContentAsString(), IdentifiableType.LINE, "line2", "v1bl2", "v1dl2", "s1", "v3bl2", "v3dl2", "s2");
         testNetworkModificationsCount(TEST_GROUP_ID, 5);
 
-        // line and switches have been removed from network and added as TombstonedEquipmentInfos in ElasticSearch
-        assertNull(network.getLine("line2"));
```
Comment (flomillot) [inline src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java:846]: Why don't we need these tests anymore ?

File: src/test/java/org/gridsuite/modification/server/Impacts/ElementImpactTest.java (hunk 6 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/modification/server/Impacts/ElementImpactTest.java:54]: Done

**Review by flomillot — CHANGES_REQUESTED** [review]

**Review by flomillot — APPROVED** [review]

**Pull request merged**